src/modules/pass.js

```diff
@@ -154,6 +154,64 @@ PassFF.Pass = (function () {
     return { item: item, quality: quality };
   }
 
+  function domainSecurityCheck(passItemURL) {
+    if (! PassFF.Preferences.autoFillDomainCheck) {
+      return true;
+    }
+    try {
+      var passURL = new URL(passItemURL);
+    } catch(e) {
+      return confirm( _("passff_error_getting_url_pass") + " (" +
+      passItemURL + ").\n" + _("passff_override_antiphishing_confirmation"));
+    }
+
+    try {
+      var currURL = new URL(window.location.href);
+    } catch(e) {
+      return confirm( _("passff_error_getting_url_curr") + " (" +
+      window.location.href + ").\n" + _("passff_override_antiphishing_confirmation"));
+    }
+
+    /*
+    Instead of requiring that the entire hostname match, which would lead to
+    example.com and login.example.com being considered different, only the
+    domains must match. However, identifying the domain is difficult because of
+    top-level-domains like .co.uk that have multiple dots in them, unlike the
+    more conventional single-dot TLDs like .com.
+
+    Resources on Identifying Domain:
+    https://stackoverflow.com/questions/10210058/get-the-parent-document-domain-without-subdomains
+    https://stackoverflow.com/questions/399250/going-where-php-parse-url-doesnt-parsing-only-the-domain
+    https://publicsuffix.org/
+
+    While not ideal, the current solution is to assume a single-dot TLD and
+    therefore match everything after the second-to-last dot. This is a security
+    risk on two-dot TLDs, as only the TLD (e.g. co.uk) will be matched.
+    */
+
+    let passDomain = passURL.hostname.split(".").slice(-2).join(".");
+    let currDomain = currURL.hostname.split(".").slice(-2).join(".");
+
+    if (passDomain != currDomain) {
+      return confirm( _("passff_domain_mismatch", [currDomain, passDomain]) +
+      "\n" + _("passff_override_antiphishing_confirmation"));
+    }
+
+    let passProt = passURL.protocol;
+    let currProt = currURL.protocol;
+
+    if (passProt != "https:") {
```

Comment:
This part below could be another function called `protocolSecurityWarning(passURL)`.
KISS :)
passItemURL -> passURL.origin